code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -1,18 +1,30 @@
 /obj/item/reagent_containers/food/drinks/cans
-	var/canopened = 0
+	var/canopened = FALSE
 	var/is_glass = 0
 	var/is_plastic = 0
+	var/shaken = 0
+	var/bursting = FALSE
+	var/burstchance = 0
 
 /obj/item/reagent_containers/food/drinks/cans/New()
 	..()
 	flags &= ~OPENCONTAINER
 
+/obj/item/reagent_containers/food/drinks/cans/examine(mob/user)
+	. = ..()
+	if(canopened)
+		. += "It has been opened."
+
 /obj/item/reagent_containers/food/drinks/cans/attack_self(mob/user)
-	if(canopened == 0)
-		playsound(loc,'sound/effects/canopen.ogg', rand(10,50), 1)
-		to_chat(user, "<span class='notice'>You open the drink with an audible pop!</span>")
-		canopened = 1
+	if(!canopened)
+		if(shaken)
+			fizzy_open(user)
+			return
+		playsound(loc, 'sound/effects/canopen.ogg', rand(10, 50), 1)
+		canopened = TRUE
 		flags |= OPENCONTAINER
+		to_chat(user, "<span class='notice'>You open the drink with an audible pop!</span>")
+		return ..()
```

Comment:
I feel like it should always call this no matter if the can is opened or not

Reply:
what exactly should be called?? return ..()???

Reply:
Indeed so. It will currently not do much to always call this since the parent just returns. But this is a bug more than anything.
`item/attack_self` sends a signal saying attack_self is called on this object. Which other things can subscribe to. So once the parent is fixed calling `..()` will ensure cans also send said signal
Btw unless specified otherwise you can assume it's about the line the comment is on :P

Reply:
Right ok, thanks. Good to know.